lms/lib/courseware_search/lms_result_processor.py

```diff
@@ -73,7 +73,10 @@ def get_display_name(category, item_id):
 
         def get_position_name(section, position):
             """ helper to fetch name corresponding to the position therein """
-            pos = int(position)
+            if position is None:
+                return None
+
+            pos = int(position.split('_', 1)[0])
```

Comment:
Can do - from it's definition it is an underscore delimited list of integers
1) It is always a string, even when it's a single number (according to its definition)
2) It doesn't need to contain an underscore - if it does not contain an underscore then the array has only one member that matches the original string
3) Again, from its definition it is a underscore delimited list of integers

I'll add a try / except to protect around it potentially changing. Python makes things very easy to do, but having no strong typing leads to needing this sort of treatment everywhere. I suppose anytime there's a cast it's a good thing to do.